### PR TITLE
Addressing Issue Number 12

### PR DESCRIPTION
**Pull request by VarthanV**
I have solved the Issue number 12 that was opened by fetching the title and the description of the website when the extension is clicked under the guidance of @xlogix

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -1,5 +1,19 @@
-var parsers = require('./lib/content/parser')
-
-var parser = parsers.NewParser()
-
-console.log(parser.getTitle(), parser.getLink())
\ No newline at end of file
+window.onload = () => {
+  var title = document.querySelector("title").text;
```
Comment (arjunmahishi) [inline src/content.js]: we can just do `document.title` right?

File: src/content.js [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+      metas[i].getAttribute("name") === "description" ||
+      metas[i].getAttribute("name") === "summary"
+    ) {
+      console.log(metas[i].getAttribute("content"));
+      data += metas[i].getAttribute("content");
+    }
+  }
+  chrome.runtime.sendMessage({
+    action: "getTitle",
+    source: `<h1> ${title} - ${data} </h1>`
```
Comment (arjunmahishi) [inline src/content.js]: we should send this data a json like:
```
{
    "title": " ",
    "description": " ",
    ...
}
```

File: src/popup/popup.js [hunk 3]
```diff
@@ -4,6 +4,28 @@ const EXISTING_PATH_SELECTOR_ID = "existing-path-selector"
...
 
+chrome.runtime.onMessage.addListener(function(request) {
+    if (request.action == "getTitle") {
+      message.innerHTML = request.source;
+    }
+  });
+
+  function onWindowLoad() {
+
+    var message = document.getElementById('message');
```
Comment (arjunmahishi) [inline src/popup/popup.js]: where is this `message` element?

File: src/popup/popup.js [hunk 4]
```diff
@@ -4,6 +4,28 @@ const EXISTING_PATH_SELECTOR_ID = "existing-path-selector"
...
+  
+    chrome.tabs.executeScript(null, {
+      file: "content.js"
```
Comment (arjunmahishi) [inline src/popup/popup.js:19]: is this the only way to inject data into popup?
if so, let's name the script something else. Something more specific and descriptive.

File: src/auth/auth.js [hunk 5]
```diff
@@ -21,9 +21,17 @@ var config = {
    *
    * When signed in, we also authenticate to the Firebase Realtime Database.
    */
+
+   var body=document.querySelector('body');
+   body.onload=function(){
+     var meta=document.querySelector('meta');
+     console.log(meta);
+   }
```
Comment (arjunmahishi) [inline src/auth/auth.js]: remove this if this was just a test. let's keep auth.js clean

File: src/auth/auth.js [hunk 6]
```diff
@@ -21,9 +21,17 @@ var config = {
...
   function initApp() {
     // Listen for auth state changes.
     // [START authstatelistener]
+    var doc= document.querySelector('meta');
+    console.log(doc);
```
Comment (arjunmahishi) [inline src/auth/auth.js]: same as above

**Review by arjunmahishi — CHANGES_REQUESTED** [review]
Apart from the comments, get rid of unwanted new lines

**Comment by VarthanV** [comment]
I have made the necessary changes  and returned a json object to the main file

**Comment by VarthanV** [comment]
No Bro what Abhishek bro said is to extract the data and send it to the backend. I am just learning GraphQL the message is nothing but a div in auth.html to make sure that my logic works . Please specify me what  i have to do after extracting the data

File: src/popup/popup.js [hunk 7]
```diff
@@ -1,27 +1,65 @@
...
-        default: showPathSelector(NEW_PATH_INPUT_ID); return
+chrome.runtime.onMessage.addListener(function(request) {
+  if (request.action == "getTitle") {
+    // This is where you get the JSON
+    var content = request.source;
+  }
+});
+
+function onWindowLoad() {
+  var message = document.getElementById("message");
```
Comment (arjunmahishi) [inline src/popup/popup.js]: This still not addressed. Can you point me to the "message" element?

File: src/popup/popup.js (hunk 4 above)
Comment (arjunmahishi) [reply src/popup/popup.js:19]: Let's rename `content.js` to something like `metadata-fetcher.js` or something similar

**Review by arjunmahishi — CHANGES_REQUESTED** [review]

**Comment by arjunmahishi** [comment]
We can merge this once the "message" element is sorted out. (this is assuming the changes are tested and working fine)

**Comment by VarthanV** [comment]
Content.js is  a convention to name the file otherwise we cannot access the dom elements and get data . I have removed the message div I just used it for debugging the data will be returned as JSON to the popup file. If you want to know whether it works i have logged the title and metadata in the console when the site loads we can disable that after you approve the PR.

**Review by arjunmahishi — APPROVED** [review]

**Comment by arjunmahishi** [comment]
@VarthanV  Thank you for your contributions!

**Pull request merged**